src/main/java/betterquesting/handlers/EventHandler.java

```diff
@@ -106,6 +108,43 @@ public void onKey(InputEvent.KeyInputEvent event) {
         }
     }
 
+
+    @SideOnly(Side.CLIENT)
+    @SubscribeEvent
+    public void onClientChatReceived(ClientChatReceivedEvent event) {
+        if (event.getMessage() != null) {
+            String text = event.getMessage().getFormattedText();
+            int index = text.indexOf("betterquesting.msg.share_quest:");
+            if (index != -1) {
+                int lastIndex = index + "betterquesting.msg.share_quest:".length();
+                int endIndex = lastIndex;
+                int questId = 0;
+                for (int i = lastIndex; i < text.length(); i++) {
+                    int digit = Character.getNumericValue(text.charAt(i));
+                    if (digit < 0) {
+                        break;
+                    }
+                    endIndex++;
+                    questId = (questId * 10) + digit;
+                }
+                IQuest quest = QuestDatabase.INSTANCE.getValue(questId);
```

Comment:
null check in case someone uses a quest ID for a quest that does not exist? In addition, maybe a unique error in this case would be warranted.